qa-tests-backend/src/test/groovy/NetworkFlowTest.groovy

```diff
@@ -475,11 +476,58 @@ class NetworkFlowTest extends BaseSpecification {
         String deploymentUid = deployments.find { it.name == EXTERNALDESTINATION }?.deploymentUid
         assert deploymentUid != null
 
-        expect:
-        "Check for edge in network graph"
+        when: "External IPs is disabled"
+        // External IPs should be disabled at this point, but it is disabled again to be safe.
+        // Later external IPs is enabled and then disabled again.
+        CollectorUtil.disableExternalIps(orchestrator)
+
         log.info "Checking for edge from ${EXTERNALDESTINATION} to external target"
         List<Edge> edges = NetworkGraphUtil.checkForEdge(deploymentUid, Constants.INTERNET_EXTERNAL_SOURCE_ID)
+        def graph = NetworkGraphService.getNetworkGraph(null, null)
+        then:
+        "There should be an edge from A to external entities and it should be the only edge"
         assert edges
+        assert edges.size() == 1
+
+        when: "External IPs is enabled"
+        CollectorUtil.enableExternalIps(orchestrator)
+        sleep 65000 // Wait for the collector scrape interval and for sensor to send connections to sensor
+
+        edges = NetworkGraphUtil.checkForEdge(deploymentUid, Constants.INTERNET_EXTERNAL_SOURCE_ID)
+        graph = NetworkGraphService.getNetworkGraph(null, null)
```

Comment:
there are 3 params why you decided to pass 2? Let's not pass anything and use defaults
```suggestion
        graph = NetworkGraphService.getNetworkGraph()
```

Reply:
Done